### PR TITLE
GitHub Action Check ES Low DPI Assets

### PR DESCRIPTION
**Pull request by samrocketman**
This automated action will clone endless-sky master branch and verify that non-`@2x` versions of files exist.  It will exit with an error if the low DPI image equivalent is not already in upstream endless-sky master.

# Please note

All pull requests which are new will fail because they won't have assets in upstream master branch, yet.  However, once the upstream pull requests get merged you can manually re-run actions which should cause them to pass.

# Tested

You can see an example run https://github.com/samrocketman/endless-sky-high-dpi/runs/8166708770?check_suite_focus=true

### CODE REVIEW

**Comment by samrocketman** [comment]
I tried a couple of things to get this PR but MCO mentions this may need to be merged before the workflow is allowed to run.  [sauce](https://discord.com/channels/251118043411775489/536900466655887360/1015613228568543263)

**Comment by samrocketman** [comment]
Also, this script was run and found issues.  The PR https://github.com/endless-sky/endless-sky-high-dpi/pull/291 resulted from this script where I verified missing assets resulted in non-zero exit code (CI failure).

File: scripts/find-low-dpi-images.sh
Comment (quyykk) [inline scripts/find-low-dpi-images.sh:10]: I don't think you need all this?

File: scripts/find-low-dpi-images.sh
Comment (quyykk) [inline scripts/find-low-dpi-images.sh:3]: This should probably be GPL licensed like everything else in ES.

File: .github/workflows/es-assets.yml
Comment (quyykk) [inline .github/workflows/es-assets.yml:4]: If we can manually trigger this action, do we need to run it automatically for PRs? Since it will always error out until the main PR is merged.

**Review by quyykk — COMMENTED** [review]
Nice!

File: scripts/find-low-dpi-images.sh
Comment (tehhowch) [inline scripts/find-low-dpi-images.sh:31]: Why do I need the script to clone ES? I have it cloned already locally. This script should take a path to the root ES directory (you can use the presence of `credits.txt` and `images/` to verify).

File: .github/workflows/es-assets.yml
Comment (tehhowch) [inline .github/workflows/es-assets.yml:1]:
```suggestion
name: Assets
```

**Review by tehhowch — CHANGES_REQUESTED** [review]
The action should be a little more than "invoke script that does everything assuming CI". At minimum, the script should work for a local user that has an existing install of the endless-sky repo without needing to download anything.

(Probably the script should not download anything at all, that's something the action can do.)

A follow-up PR can amend the updated script to handle args for specific image. Adding an action to work on PRs with a particular upstream dependency can follow.

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh:10]: I add the development environment in which scripts were created since tools change

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh:3]: I typically license my code MIT which is compatible with GPL3.

File: .github/workflows/es-assets.yml
Comment (samrocketman) [reply .github/workflows/es-assets.yml:4]: I defer to this repo to adjust whatever is necessary.  It can be removed if you want

Comment (samrocketman) [reply .github/workflows/es-assets.yml:4]: As I said this is an initial pass.  Parsing pull request description can be added later.  Better to start somewhere than have nothing.

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh:31]: This is for an action not your local clone.  It can be updated later to work from local
if you want.  i.e. take an arg.

File: .github/workflows/es-assets.yml
Comment (samrocketman) [reply .github/workflows/es-assets.yml:1]: This suggestion is less descriptive.

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh:37]: This is an aesthetic request which actually breaks how it handless nonzero.  I had this to start and it misreported nonzero when it should have been zero.  I suggest you take what I have which is tested and works.  I won't be committing this into this PR.

File: scripts/find-low-dpi-images.sh
Comment (tehhowch) [reply scripts/find-low-dpi-images.sh:31]: Why should we add a script that unnecessarily downloads the ES repo when only the CI environment needs that bit of functionality?

**Comment by samrocketman** [comment]
I went ahead and rewrote it and relicensed it under GPL3.  You can provide a local clone of endless-sky as the first argument and it will use that, instead.

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh]: Then don't.  You want to have something that does it different you're welcome to provide it.  It was pointed out on Discord that a feature like this was desirable.  It's typical for a CI environment to be able to clone what it's checking against.  I defer to you.

Definitively close this if you don't want it.  I've made the changes I'm going to make for now and anything extra can come later or you can just full reject it.  Up to you.

Comment (samrocketman) [reply scripts/find-low-dpi-images.sh]: Perhaps it would make more sense for me to explain why I think your requests in this instance didn't make sense to me.

* The majority of those contributing to this repository are on Windows.
* Of the minority who may be on Mac; a small percentage would be technical to fire up a shell and run this script.

This script is designed for CI and intended to be run within CI.

File: .github/workflows/es-assets.yml
Comment (samrocketman) [inline .github/workflows/es-assets.yml:6]:
```suggestion
```

File: scripts/find-low-dpi-images.sh
Comment (samrocketman) [reply scripts/find-low-dpi-images.sh]: More specifically someone running from Mac (BSD tools) should understand this was developed from Linux (GNU or findutils tools).  There may be changes necessary for a Mac user.

While the version on Mac bash is very old (v3 I think) I am using bash features available to it.

**Comment by samrocketman** [comment]
This seems to be stuck in analysis paralysis.

Friendly reminder that this gets us 90% there in terms of need.  Checking a pull request against upstream master as a quick sanity check before merging asset mistakes.

The remaining 10% requested (parsing against upstream pull requests) is a non-trivial amount of work and not something I'm inclined to do now as part of this PR.

I've updated the license upon request and even updated some of the code style requests which were aesthetic and didn't change functionality.

What's the verdict here?  If you still don't want it then I say let's close it and move on as I'm tired of wasting my time checking in on this PR.

**Comment by samrocketman** [comment]
Seems there’s little desire to incorporate this.

**Pull request closed without merging**
